### PR TITLE
Update keymanager templates and parameters

### PR DESCRIPTION
**Pull request by Savidude**

### CODE REVIEW

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 [hunk 1]
```diff
@@ -409,11 +409,11 @@
             <!-- Keystore file location-->
             <Location>{{ internal_keystore_location }}</Location>
             <!-- Keystore type (JKS/PKCS12 etc.)-->
-            <Type>{{ internal_keystore_type }}</Type>
+            <Type>JKS</Type>
             <!-- Keystore password-->
-            <Password>{{ internal_keystore_password }}</Password>
+            <Password>wso2carbon</Password>
```
Comment (VimukthiPerera) [inline roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: Pamatereize this as well.

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 [hunk 2]
```diff
@@ -409,11 +409,11 @@
...
             <!-- Private Key alias-->
-            <KeyAlias>{{ internal_keystore_key_alias }}</KeyAlias>
+            <KeyAlias>wso2carbon</KeyAlias>
```
Comment (VimukthiPerera) [inline roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: Need to parameterize this.

File: roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2 [hunk 3]
```diff
@@ -27,7 +27,7 @@
             <!-- By default users in this role sees the registry root -->
             <Property name="isCascadeDeleteEnabled">true</Property>
             <Property name="initializeNewClaimManager">true</Property>
-            <Property name="dataSource">jdbc/WSO2CarbonDB</Property>
+            <Property name="dataSource">{{ datasource }}</Property>
```
Comment (VimukthiPerera) [inline roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2]: This should be `WSO2UM_DB`. No need to parameterize this

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 (hunk 1 above)
Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: +1 will have to parameterize this for Gateway as well

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 (hunk 2 above)
Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: +1 will parameterize KeyStore ad InternalKeyStore for both Gateway and KM

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 (hunk 1 above)
Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: Added https://github.com/wso2/ansible-apim/pull/49/commits/f26ff3b4bf62ab81d6a89ce61855141129c2690f

File: roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2 (hunk 2 above)
Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/carbon.xml.j2]: Added https://github.com/wso2/ansible-apim/pull/49/commits/f26ff3b4bf62ab81d6a89ce61855141129c2690f

File: roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2 (hunk 3 above)
Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2]: Added https://github.com/wso2/ansible-apim/pull/49/commits/f26ff3b4bf62ab81d6a89ce61855141129c2690f

File: dev/group_vars/apim.yml [hunk 4]
```diff
@@ -111,16 +98,31 @@ wso2_mb_store_db_username: wso2carbon
...
 trust_store_password: wso2carbon
 
 # The KeyStore which is used for encrypting/decrypting internal data. This block is read by Carbon Crypto Service
 internal_keystore_location: ${carbon.home}/repository/resources/security/wso2carbon.jks
 internal_keystore_key_password: wso2carbon
+internal_keystore_password: wso2carbon
+internal_keystore_key_alias: wso2carbon
 
 # User Manager configurations
 datasource: jdbc/WSO2CarbonDB
```
Comment (VimukthiPerera) [inline dev/group_vars/apim.yml]: Shall we remove this?

File: roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2 [hunk 5]
```diff
@@ -27,7 +27,7 @@
             <!-- By default users in this role sees the registry root -->
             <Property name="isCascadeDeleteEnabled">true</Property>
             <Property name="initializeNewClaimManager">true</Property>
-            <Property name="dataSource">jdbc/WSO2CarbonDB</Property>
+            <Property name="dataSource">WSO2_UM_DB</Property>
```
Comment (VimukthiPerera) [inline roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2]: This should be `jdbc/WSO2UM_DB`.

Comment (Savidude) [reply roles/apim-km/templates/carbon-home/repository/conf/user-mgt.xml.j2]: Fixed https://github.com/wso2/ansible-apim/pull/49/commits/6464f58613fc352dd7b9fcc50d0bb1c1f9d8a6e1

File: dev/group_vars/apim.yml (hunk 4 above)
Comment (Savidude) [reply dev/group_vars/apim.yml]: Done https://github.com/wso2/ansible-apim/pull/49/commits/6464f58613fc352dd7b9fcc50d0bb1c1f9d8a6e1

**Pull request merged**